deprecations_to_md.py

```diff
@@ -21,10 +21,10 @@
     f.write(Path(header).read_text())
     for (ver, features) in sorted(by_version.items()):
         f.write(f"""
-### Features available up to and including --simics-api={ver.str}
+### Features available up to and including -\\-simics-api={ver.str}
```

Comment:
oh, ew. I won't block this, but I wonder if this is ugly enough that it would warrant using a raw string instead, and accomplishing formatting via `.format` instead.